File: bindings/java-jni/src/main/java/org/maplibre/nativejni/map/MapMode.java
package org.maplibre.nativejni.map;

/** Map rendering mode selected at map creation. */
public enum MapMode {
  CONTINUOUS(0),
  STATIC(1),
  TILE(2);

  private final int nativeValue;

  MapMode(int nativeValue) {
    this.nativeValue = nativeValue;
  }

  public int nativeValue() {
    return nativeValue;
  }
}

File: bindings/dotnet/src/Maplibre.Native/Map/MapMode.cs
namespace Maplibre.Native.Map;

/// <summary>Map rendering mode.</summary>
public enum MapMode : uint
{
    /// <summary>The map continuously renders when invalidated.</summary>
    Continuous = 0,

    /// <summary>The map renders still images on request.</summary>
    Static = 1,

    /// <summary>The map renders tile-sized outputs on request.</summary>
    Tile = 2,
}
